File: java-backoffice/src/main/java/br/com/unialfa/model/Vaga.java
package br.com.unialfa.model;

public class Vaga {

    // Atributos privados
    private Long id;
    private String titulo;
    private String descricao;
    private Double bolsa; // Valor da remuneração / bolsa auxílio
    private Long empresaId; // Guarda qual empresa é a dona da vaga
    private boolean ativa;

    // Construtor vazio
    public Vaga() {
    }

    // Construtor completo
    public Vaga(Long id, String titulo, String descricao, Double bolsa, Long empresaId, boolean ativa) {
        this.id = id;
        this.titulo = titulo;
        this.descricao = descricao;
        this.bolsa = bolsa;
        this.empresaId = empresaId;
        this.ativa = ativa;
    }

    // --- Getters e Setters (Encapsulamento) ---

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public Double getBolsa() { return bolsa; }
    public void setBolsa(Double bolsa) { this.bolsa = bolsa; }

    public Long getEmpresaId() { return empresaId; }
    public void setEmpresaId(Long empresaId) { this.empresaId = empresaId; }

    public boolean isAtiva() { return ativa; }
    public void setAtiva(boolean ativa) { this.ativa = ativa; }
}

File: php-frontend/src/Models/Vaga.php
<?php
// Representa uma vaga de estágio no portal.
// Campos do banco (portal_estagios.vagas):
// id          → bigint(20)
// titulo      → varchar(255)
// descricao   → text
// bolsa       → decimal(10,2) — valor da bolsa em R$
// empresa_id  → bigint(20)    — FK para empresas
// ativa       → tinyint(1)    — 1 = aberta, 0 = encerrada

class Vaga {

    // ── Atributos privados (encapsulamento) ──
    private int    $id;
    private string $titulo;
    private string $descricao;
    private float  $bolsa;      // decimal vira float no PHP
    private int    $empresaId;  // campo no banco: empresa_id
    private bool   $ativa;      // tinyint vira bool no PHP

    // ── Construtor ──
    // Recebe array com dados vindos da API
    // Ex: ['id' => 1, 'titulo' => 'Dev Web', 'bolsa' => 800.00, 'ativa' => 1]
    public function __construct(array $data) {
        $this->id        = $data['id']         ?? 0;
        $this->titulo    = $data['titulo']     ?? '';
        $this->descricao = $data['descricao']  ?? '';
        // decimal(10,2) vem como string da API — converte para float
        $this->bolsa     = (float)($data['bolsa']      ?? 0);
        // banco usa empresa_id com underscore
        $this->empresaId = (int)($data['empresa_id']   ?? 0);
        // tinyint(1) vem como 0 ou 1 da API — converte para bool
        $this->ativa     = (bool)($data['ativa']       ?? true);
    }

    // ── Getters ──
    public function getId(): int        { return $this->id; }
    public function getTitulo(): string { return $this->titulo; }
    public function getDescricao(): string { return $this->descricao; }
    public function getBolsa(): float   { return $this->bolsa; }
    public function getEmpresaId(): int { return $this->empresaId; }
    public function isAtiva(): bool     { return $this->ativa; }

    // ── Métodos auxiliares ──
    // Formata a bolsa em Real brasileiro
    // Ex: 800.00 → R$ 800,00
    public function getBolsaFormatada(): string {
        return 'R$ ' . number_format($this->bolsa, 2, ',', '.');
    }

    public function getStatusLabel(): string {
        return $this->ativa ? 'Aberta' : 'Encerrada';
    }

    public function getStatusBadge(): string {
        return $this->ativa ? 'badge-green' : 'badge-gray';
    }
}
